server/src/main/java/io/crate/execution/ddl/tables/AddColumnTask.java

```diff
@@ -101,6 +100,19 @@ protected ClusterState execute(ClusterState currentState, AddColumnRequest reque
 
         request = addMissingParentColumns(request, currentTable);
         HashMap<ColumnIdent, List<Reference>> tree = buildTree(request.references());
+
+        // Request can contain parent columns of the new columns in order to be able to build and traverse the tree.
+        // For incrementing column OID we only count new columns which is basically equivalent of counting leaves.
+        long leaves = 0;
+        for (Reference reference: request.references()) {
+            // Ref is a leaf if it doesn't have children.
+            if (tree.get(reference.column()) == null) {
+                leaves++;
+            }
+        }
+        Metadata.Builder metadataBuilder = Metadata.builder(currentState.metadata());
+        metadataBuilder.columnOID(currentState.metadata().columnOID() + leaves);
```

Comment:
this covers 
1. adding single/multiple column via ALTER TABLE ADD COLUMN
2. adding columns via dynamic mapping update in the new indexer (as it fires `AddColumnRequest`-s)